provider/marathon_test.go

```diff
@@ -519,57 +519,19 @@ func TestMarathonTaskFilter(t *testing.T) {
 		{
 			task: marathon.Task{
 				AppID: "specify-port-number",
-				Ports: []int{80, 443},
+				Ports: []int{8080},
 			},
 			applications: &marathon.Applications{
 				Apps: []marathon.Application{
 					{
 						ID:    "specify-port-number",
 						Ports: []int{80, 443},
 						Labels: &map[string]string{
-							"traefik.port": "80",
-						},
-					},
-				},
-			},
-			expected:         true,
-			exposedByDefault: true,
-		},
-		{
-			task: marathon.Task{
-				AppID: "specify-unknown-port-number",
-				Ports: []int{80, 443},
-			},
-			applications: &marathon.Applications{
-				Apps: []marathon.Application{
-					{
-						ID:    "specify-unknown-port-number",
-						Ports: []int{80, 443},
-						Labels: &map[string]string{
 							"traefik.port": "8080",
 						},
 					},
 				},
 			},
-			expected:         false,
-			exposedByDefault: true,
-		},
-		{
-			task: marathon.Task{
-				AppID: "specify-port-index",
-				Ports: []int{80, 443},
-			},
-			applications: &marathon.Applications{
-				Apps: []marathon.Application{
-					{
-						ID:    "specify-port-index",
```

Comment:
The `processPorts()` function [returns a non-empty list of ports if there are task ports](https://github.com/nicgrayson/traefik/blob/2d3f451c8aec9bd8739fc3e5170df7280151fd6d/provider/marathon.go#L477-L479). Since we define task ports during our test fixture, we should be able to make the test pass even without the application ports, shouldn't we?

In fact, I don't see application ports being used anywhere in the production code, so I'm suspecting they are a relict that could be removed from all tests. For now though, I'd prefer them to be removed from the tests we touch and leave a possibly wider cleanup to a separate PR.